src/bridge/cockpitcpusamples.c

```diff
@@ -71,28 +71,45 @@ cockpit_cpu_samples (CockpitSamples *samples)
       guint64 system;
       guint64 idle;
       guint64 iowait;
+      gchar cpu_core[8];
+      gchar core[5];
 
-      if (!(g_str_has_prefix (line, "cpu ")))
+      if (!(g_str_has_prefix (line, "cpu")))
         continue;
 
       #define FMT64 "%" G_GUINT64_FORMAT " "
-      if (sscanf (line + sizeof ("cpu ") - 1, FMT64 FMT64 FMT64 FMT64 FMT64,
+      if (sscanf (line, "%s " FMT64 FMT64 FMT64 FMT64 FMT64,
+                  cpu_core,
                   &user,
                   &nice,
                   &system,
                   &idle,
-                  &iowait) != 5)
+                  &iowait) != 6)
         {
           g_warning ("Error parsing line %d of /proc/stat with content `%s'", n, line);
           continue;
         }
 
       user_hz = ensure_user_hz ();
-      cockpit_samples_sample (samples, "cpu.basic.nice", NULL, nice*1000/user_hz);
-      cockpit_samples_sample (samples, "cpu.basic.user", NULL, user*1000/user_hz);
-      cockpit_samples_sample (samples, "cpu.basic.system", NULL, system*1000/user_hz);
-      cockpit_samples_sample (samples, "cpu.basic.iowait", NULL, iowait*1000/user_hz);
-      break;
+      if (strlen (cpu_core) > 3)
+        {
+          if (sscanf (cpu_core, "cpu%s", core) != 1)
+            {
+              g_warning ("Error parsing core /proc/stat with content `%s'", cpu_core);
+              continue;
+            }
+          cockpit_samples_sample (samples, "cpu.core.nice", core, nice*1000/user_hz);
+          cockpit_samples_sample (samples, "cpu.core.user", core, user*1000/user_hz);
+          cockpit_samples_sample (samples, "cpu.core.system", core, system*1000/user_hz);
+          cockpit_samples_sample (samples, "cpu.core.iowait", core, iowait*1000/user_hz);
+        }
+      else
+        {
+          cockpit_samples_sample (samples, "cpu.basic.nice", NULL, nice*1000/user_hz);
+          cockpit_samples_sample (samples, "cpu.basic.user", NULL, user*1000/user_hz);
+          cockpit_samples_sample (samples, "cpu.basic.system", NULL, system*1000/user_hz);
+          cockpit_samples_sample (samples, "cpu.basic.iowait", NULL, iowait*1000/user_hz);
```

Comment:
If you feel like it, please add a smoketest to src/bridge/test-metrics.c similar to `test_cgroup` -- you can check that the calculated rate is in a sensible range between 0 and 1000 (msec/s). This is usually a great helper for development, but that part is already done now. :grin: I'm also happy to send a PR for that as a follow-up (I really like writing these)